src/viking.js

```diff
@@ -1,11 +1,63 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health;
+        this.strength = strength;
+    }
+
+    attack() {
+        return this.strength;
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage;
+    }    
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength);
+        this.name = name;
+    }
+    receiveDamage(damage) {
+        this.health -= damage;
+        if (this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`
+        } else {
+            return `${this.name} has died in act of combat`
+        }
+    }
+    battleCry() {
+        return "Odin Owns You All!";
+    }
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+    receiveDamage(damage) {
+        this.health -=damage;
+        if (this.health > 0) {
+            return `A Saxon has received ${damage} points of damage`
+        } else {
+            return `A Saxon has died in combat`
+        }
+        }
+}
 
 // War
-class War {}
+class War {
+    constructor() {
+        vikingArmy: [],
+        saxonArmy: []
+    }
+    addViking(viking) {
+        this.vikingArmy = 1;
+    } 
+    addSaxon(saxon) {
+        this.saxonArmy = 1;
+    }
+    vikingAttack()
```

Comment:
If you don't see your tests passing still, just comment this three functions that are not yet written in order to see the tests passing (if you want to check on your code).